### PR TITLE
Native Go SDK publishing

### PR DESCRIPTION
**Pull request by danielrbradley**
Part of https://github.com/pulumi/ci-mgmt/issues/915

Implement Go SDK publishing for native providers 

- Add allowed changes to CheckWorktreeClean.
- Add PublishGoSdk step & job.
- Give version step an id by default.
- Switch release workflow to use PublishGoSdk instead of TagSdk.

Remove azure-native as it's never been managed by this tool so there's no reason to generate it here.

Replace `pulumictl get version` for goreleaser step with version action used everywhere else.

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -62,6 +62,7 @@ jobs:
     timeout-minutes: 240
     runs-on: ubuntu-latest
     needs: deploy
+    if: needs.deploy.outputs.pull_request_created == 'true'
```
[the remark this answers is not in the document]
Comment (danielrbradley) [reply .github/workflows/pull-request.yml:65]: If the PR creation fails for any reason, we still don't want to wait for a non-existant PR.

I'm not following what regressions this is going to miss - if there's no changes to be pushed, surely there's no chance of regressions?

[the remark this answers is not in the document]
Comment (VenelinMartinov) [reply .github/workflows/pull-request.yml:65]: If the PR creation fails for any reason, the ci-mgmt check should fail too

[the remark this answers is not in the document]
Comment (VenelinMartinov) [reply .github/workflows/pull-request.yml:65]: If makes a change to ci-mgmt which makes the PR creation fail in providers, the xyz check should catch that.

**Review by guineveresaenger — APPROVED** [review]
This looks like what's already working over here in TF-bridged land.

**Pull request merged**